osh/cmd_eval.py

```diff
@@ -2068,8 +2074,8 @@ def _MaybeRunErrTrap(self):
             return
 
         # bash rule - affected by set -o errtrace
-        if self.mem.InsideFunction():
-            return
+        # if self.mem.InsideFunction():
```

Comment:
why comment this out?  I'm surprised it didn't make a test fail ...

But it also didn't make `builtin-trap-err` pass